notebooks/mitocheck-map-plotting.py

```diff
@@ -0,0 +1,427 @@
+reg_dp_sc_mAPs["feature_type"] = "DP"
+shuffled_feat_dp_sc_mAPs = dp_sc_mAPs.loc[dp_sc_mAPs["shuffled"] == "features_shuffled"]
+shuffled_feat_dp_sc_mAPs["feature_type"] = "DP"
+shuffled_pheno_dp_sc_mAPs = dp_sc_mAPs.loc[
+    dp_sc_mAPs["shuffled"] == "phenotype_shuffled"
+]
+shuffled_pheno_dp_sc_mAPs["feature_type"] = "DP"
+
+reg_cp_dp_sc_mAPs = cp_dp_sc_mAPs.loc[cp_dp_sc_mAPs["shuffled"] == "non-shuffled"]
+reg_cp_dp_sc_mAPs["feature_type"] = "CP_DP"
+shuffled_feat_cp_dp_sc_mAPs = cp_dp_sc_mAPs.loc[
+    cp_dp_sc_mAPs["shuffled"] == "features_shuffled"
+]
+shuffled_feat_cp_dp_sc_mAPs["feature_type"] = "CP_DP"
+shuffled_pheno_cp_dp_sc_mAPs = cp_dp_sc_mAPs.loc[
+    cp_dp_sc_mAPs["shuffled"] == "phenotype_shuffled"
+]
+shuffled_pheno_cp_dp_sc_mAPs["feature_type"] = "CP_DP"
+
+
+# In[5]:
+
+
+# Generating sampling_error df
+# This table will be used to merge with the aggregate table to get the sampling error a specific category.
+merged_sc_ap_scores_df = pd.concat(
+    [
+        reg_cp_sc_mAPs,
+        reg_dp_sc_mAPs,
+        shuffled_feat_cp_sc_mAPs,
+        shuffled_pheno_cp_sc_mAPs,
+        shuffled_feat_dp_sc_mAPs,
+        shuffled_pheno_dp_sc_mAPs,
+        reg_cp_dp_sc_mAPs,
+        shuffled_feat_cp_dp_sc_mAPs,
+        shuffled_pheno_cp_dp_sc_mAPs,
+    ]
+)
+
+df_group = merged_sc_ap_scores_df.groupby(
+    by=["Mitocheck_Phenotypic_Class", "feature_type", "shuffled"]
+)
+
+sampling_error_df = []
+for name, df in df_group:
+    pheno, feature_type, shuffled_type = name
+
+    # caclulating sampling error
+    avg_percision = df["average_precision"].values
+    sampling_error = np.std(avg_percision) / np.sqrt(len(avg_percision))
+
+    sampling_error_df.append([pheno, feature_type, shuffled_type, sampling_error])
+cols = ["Mitocheck_Phenotypic_Class", "feature_type", "shuffled", "sampling_error"]
+sampling_error_df = pd.DataFrame(sampling_error_df, columns=cols)
+sampling_error_df
+
+# updating name:
+sampling_error_df.loc[
+    sampling_error_df["shuffled"] == "phenotype_shuffled"
+] = "phenotypes_shuffled"
+
+
+# In[6]:
+
+
+# aggregate single cells scores with cell UUID
+data = tuple(merged_sc_ap_scores_df.groupby(by=["Cell_UUID"]))
+columns = merged_sc_ap_scores_df.columns
+agg_sc_ap_scores_df = []
+for cell_id, df1 in data:
+    for feature_type, df2 in df1.groupby(by="feature_type"):
+        for shuffle_type, df3 in df2.groupby(by="shuffled"):
+            aggregated_ap_score = df3["average_precision"].mean()
+
+            selected_row = df3.iloc[0]
+            selected_row["average_precision"] = aggregated_ap_score
+            agg_sc_ap_scores_df.append(selected_row.values.tolist())
+
+# saving into the results repo
+agg_sc_ap_scores_df = pd.DataFrame(data=agg_sc_ap_scores_df, columns=columns)
+agg_sc_ap_scores_df.to_csv(
+    sc_ap_scores_dir / "merged_sc_agg_ap_scores.csv", index=False
+)
+agg_sc_ap_scores_df
+
+
+# In[7]:
+
+
+# Generating aggregate scores with a threshold p-value of 0.05
```

Comment:
is this alpha adjusted??? Is it adjusted for multiple comparisons later??

Reply:
I do not know. Looking at the code, it seems it does:
https://github.com/cytomining/copairs/blob/91a282b9b0fa7a239469b3b96e198345ac5a108e/src/copairs/map.py#L64